src/NextGen/identity/identity.ts

```diff
@@ -0,0 +1,31 @@
+import { MPID, User } from '@mparticle/web-sdk';
+import { IMParticleComponent } from '../component';
+import { IdentityCallback } from './identityCallback';
+
+// QUESTION: What is the diff between this and IdentityApiData?
+// TODO: sync this with android class
+export interface IdentityApiResult {}
+export interface IdentityApiRequest {}
+
+// QUESTION: Should we mirror the way Android handles HTTP codes, or do it the way we've been doing it?
+// https://docs.mparticle.com/developers/sdk/android/idsync/#idsync-status-codes
+// https://docs.mparticle.com/developers/sdk/web/idsync/#error-handling
+
+export interface IMparticleIdentity extends IMParticleComponent {
+    // QUESTION: Should these be callbacks or promises that eventually return something?
+    login(callback: IdentityCallback): void;
+    // login(): IdentityApiResult;
+    logout(callback: IdentityCallback): void;
+    // logout(): IdentityApiResult;
+    getUser(mpid: MPID, callback: IdentityCallback): User;
+    getUsers(callback: IdentityCallback): User[];
```

Comment:
Neither of these make async requests, so I'm curious why they have callbacks.  Probably a question for @markvdouw

Reply:
From conversation: Prefer async/await, fail back to callbacks if we can.

Reply:
Making them Promises for now. We should make it a point to test backwards compatibility once they have been fully implemented.